src/Arcus.EventGrid.Tests.Integration/Publishing/EventPublishingTests.cs

```diff
@@ -191,6 +222,80 @@ await EventGridPublisherBuilder
                 });
         }
 
+        [Fact]
+        public async Task PublishMultipleRawEvents_WithBuilder_ValidParameters_SucceedsWithRetryCount()
+        {
+            // Arrange
+            var topicEndpoint = Configuration.GetValue<string>("Arcus:EventGrid:TopicEndpoint");
+            var endpointKey = Configuration.GetValue<string>("Arcus:EventGrid:EndpointKey");
+            const string licensePlate = "1-TOM-1337";
+            var events =
+                Enumerable
+                    .Repeat<Func<Guid>>(Guid.NewGuid, 2)
+                    .Select(newGuid => new RawEvent(
+                                newGuid().ToString(),
+                                subject: "integration-test",
+                                body: $"{{\"licensePlate\": \"{licensePlate}\"}}",
+                                type: "Arcus.Samples.Cars.NewCarRegistered",
+                                dataVersion:"1.0",
+                                eventTime: DateTimeOffset.Now))
+                    .ToArray();
+
+            // Act
+            await EventGridPublisherBuilder
+                  .ForTopic(topicEndpoint)
+                  .UsingAuthenticationKey(endpointKey)
+                  .Build()
+                  .PublishManyAsync(events);
+
+            // Assert
+            Assert.All(
+                events,
+                rawEvent =>
+                {
+                    TracePublishedEvent(rawEvent.Id, events);
+                    string receivedEvent = _serviceBusEventConsumerHost.GetReceivedEvent(rawEvent.Id, 5);
+                    AssertReceivedEvent(rawEvent.Id, rawEvent.EventType, rawEvent.Subject, licensePlate, receivedEvent);
```

Comment:
Same for other fields, you are asserting on things that are 99.99% probably the same as you are basing yourself on returned values.

You should use all initial values to make sure that the assert is real.